guilib/measure/camera/badpixels.py

```diff
@@ -1354,42 +1354,44 @@ def write(self, filepath=''):
-        repl_width = max(len(row) for row in columns[1])
-
-        # And format the lines
-        columns[0][0] = f"#{columns[0][0]:>{bad_width-1}}"
-        columns[0][1:] = [f"{row:>{bad_width}}" for row in columns[0][1:]]
-        columns[1] = [f"{row:>{repl_width}}" for row in columns[1]]
-        lines = '\n'.join(f'{bad}, {repl}' for bad, repl in zip(*columns))
-
-        filepath = Path(filepath)
-        if not filepath.exists():
-            filepath.mkdir(parents=True)
-        filename = filepath / f"{self.__base_name}.badpx"
-
-        width, height, *_ = self.__camera.image_info
-        n_bad = self.n_bad_pixels_sensor
-        n_uncorr = self.n_uncorrectable_sensor
-        comment = (f"# Bad pixels file for camera {self.__camera.name}. \n"
-                   f"# Total number of bad pixels: {n_bad}, i.e., "
-                   f"{100*n_bad/(width*height):.2f}% of the sensor.\n")
-        if n_uncorr:
-            comment += f"# Of these, {n_uncorr} cannot be corrected.\n"
-        comment += ("# Bad pixel coordinates are (row number, column "
-                    "number), i.e., (y, x).\n#\n")
-
+        with np.printoptions(legacy='1.25'):
+            # Prepare the column contents
+            columns = [
+                ['Bad pixels',
+                 *(f"{tuple(b)}" for b in self.__bad_coords),
+                 *(f"{tuple(b)}" for b in self.__uncorrectable)],
+                ['Replacements',
+                 *(f"{tuple(r)}" for r in self.__replacements),
+                 *(['NaN']*len(self.__uncorrectable))]
+                ]
+
+            # Find padding lengths:
+            bad_width = max(len('# Bad pixels'),
+                            *(len(row) for row in columns[0]))
+            repl_width = max(len(row) for row in columns[1])
+
+            # And format the lines
+            columns[0][0] = f"#{columns[0][0]:>{bad_width-1}}"
+            columns[0][1:] = [f"{row:>{bad_width}}" for row in columns[0][1:]]
+            columns[1] = [f"{row:>{repl_width}}" for row in columns[1]]
+            lines = '\n'.join(f'{bad}, {repl}' for bad, repl in zip(*columns))
+
+            filepath = Path(filepath)
+            if not filepath.exists():
+                filepath.mkdir(parents=True)
+            filename = filepath / f"{self.__base_name}.badpx"
+
+            width, height, *_ = self.__camera.image_info
+            n_bad = self.n_bad_pixels_sensor
+            n_uncorr = self.n_uncorrectable_sensor
+            comment = (f"# Bad pixels file for camera {self.__camera.name}. \n"
+                       f"# Total number of bad pixels: {n_bad}, i.e., "
+                       f"{100*n_bad/(width*height):.2f}% of the sensor.\n")
+            if n_uncorr:
+                comment += f"# Of these, {n_uncorr} cannot be corrected.\n"
+            comment += ("# Bad pixel coordinates are (row number, column "
+                        "number), i.e., (y, x).\n#\n")
+        
```

Comment:
I believe this bit can stay non indented. Looks like the array-to-string part only happens when columns is created. The rest of the formatting uses string values already.